src/pages/index.js

```diff
@@ -0,0 +1,307 @@
+//     cards.forEach((item) => {
+//       renderCard(item, "append");
+//     });
+//   })
+//   .catch(console.error);
+
+  // Iterates over Initial Card Array
+// initialCards.forEach((item) => {
+//   renderCard(item, "append");
+// });
+
+
+// Connects the individual parts of the card elements to the template
+function getCardElement(data){
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button")
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardNameEl.textContent = data.name;
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardLikeButton.addEventListener("click", (evt) => handleLikeCard(evt, data._id));
+  cardDeleteButton.addEventListener("click", () => {handleDeleteCard(cardElement, data._id)});
+
+  cardImage.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    prewviewModalCaptionEl.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+function renderCard(item, method = "prepend") {
+  const cardElement = getCardElement(item);
+  cardsList[method](cardElement);
+}
+
+// Open Modal
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+// Close Modal
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+
+// Escape Key Close
+function handleEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    const currentModal = document.querySelector(".modal_opened");
+    if (currentModal) {
+      closeModal(currentModal);
+    }
+  }
+}
+
+// Iterate over modal overlay to add click outside of content
+overlays.forEach((modal) => {
+  modal.addEventListener("click", (evt) => {
+    if (evt.target === modal && modal.classList.contains("modal_opened")) {
+      closeModal(modal);
+    }
+  });
+});
+
+// Edit Profile
+// ---------------------------
+function handleEditFormSubmit(evt) {
+  handleSubmit(updateEditProfileForm, evt, 'Saving. . .', settings.inactiveButtonClass);
+}
+
+function updateEditProfileForm() {
+
+  return api.editUserInfo({
+    name: editModalNameInput.value,
+    about: editModalDescriptionInput.value
+  })
+    .then((userInfo) => {
+
+      profileName.textContent = userInfo.name;
+      profileDescription.textContent = userInfo.about;
+      closeModal(editProfileModal);
+      // disableButton(editFormElement.querySelector(settings.submitButtonSelector), settings.inactiveButtonClass)
+
+    })
+    // .then(() => closeModal(editProfileModal));
+    .catch(console.error)
```

Comment:
`.catch(console.error)` is not needed if you use `handleSubmit`, because it's already inside it

Reply:
* Another remark hasn't been corrected https://github.com/jhouse233/se_project_spots/pull/1#discussion_r1978273346 `.catch(console.error)` is not needed if you use `handleSubmit`, because it's already inside it. Please, delete it from lne 183 (from `updateEditProfileForm`)